File: TwoSum.java
import java.util.HashMap;
import java.util.Map;

public class TwoSum {

    public static void main(String[] args) {
        TwoSum twoSum = new TwoSum();
        int[] nums = { 2, 7 };
        int target = 9;
        int[] result = twoSum.twoSumTwoPointers(nums, target);
        System.out.print(result[0] + ", " + result[1]);
    }

    public int[] twoSumTwoPointers(int[] nums, int target) {

        int[] result = new int[2];

        int i = 0;
        int j = nums.length - 1;

        while (i < j) {
            int sum = nums[i] + nums[j];
            if (sum == target) {
                result[0] = i + 1;
                result[1] = j + 1;
                break;
            } else if (sum < target) {
                i++;
            } else {
                j--;
            }
        }

        return result;
    }

    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            map.put(target - nums[i], i);
        }
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(nums[i])) {
                result[0] = i + 1;
                result[1] = map.get(nums[i]) + 1;
                break;
            }
        }
        return result;
    }
}

File: TwoSum.js
//given a sorted array return two elements that has the sum equal to target value k
function twoSum(nums, k) {
    var start = 0
    var end = nums.length - 1
    var count = 0;
    while (start < end) {
        if (nums[start] + nums[end] == k) {
            count++
            start++
            end--
        } else if (nums[start] + nums[end] > k) {
            end--
        } else {
            start++
        }
    }
    return count
}


console.log(twoSum([1, 2, 3, 4, 5], 8)); // [3, 5]
console.log(twoSum([1, 4, 4, 5, 5, 5, 6, 6, 11], 11)); // [3, 5]
